valhalla/filesystem.h

```diff
@@ -376,9 +376,9 @@ inline bool create_directories(const path& p) {
     if (stat(partial.c_str(), &s) != 0) {
       // create this piece with filesystem::permissions::all
 #ifdef _WIN32
-      if (_mkdir(partial.c_str()) != 0) {
+      if (_mkdir(partial.c_str()) != 0 && errno != EEXIST) {
 #else
-      if (mkdir(partial.c_str(), S_IRWXU | S_IRWXG | S_IRWXO) != 0) {
+      if (mkdir(partial.c_str(), S_IRWXU | S_IRWXG | S_IRWXO) != 0 && errno != EEXIST) {
 #endif
         return false;
```

Comment:
we only give up if we failed to create a sub directory in the tree AND it was for some other reason than the directory already existed.